pkg/asset/machines/aws/machines.go

```diff
@@ -255,6 +258,9 @@ func provider(in *machineProviderInput) (*machineapi.AWSMachineProviderConfig, e
 		CredentialsSecret: &corev1.LocalObjectReference{Name: "aws-cloud-credentials"},
 		Placement:         machineapi.Placement{Region: in.region, AvailabilityZone: in.zone},
 		SecurityGroups:    securityGroups,
+		CPUOptions: &machineapi.CPUOptions{
+			ConfidentialCompute: ptr.To(machineapi.AWSConfidentialComputePolicy(*in.cpuOptions.ConfidentialCompute)),
```

Comment:
unit tests are failing because of a panic. You need to check if in.cpuOptions is set